### PR TITLE
fix get_output_dim() for ElmoTokenEmbedder

### PR DESCRIPTION
**Pull request by tingkai-zhang**
1. Fix get_output_dim() of ElmoTokenEmbedder with projection
2. Add test of get_output_dim() of ElmoTokenEmbedder with projection
 
#2227

### CODE REVIEW

File: allennlp/tests/modules/token_embedders/elmo_token_embedder_test.py [hunk 1]
```diff
@@ -87,10 +87,12 @@ def test_forward_works_with_projection_layer(self):
         word2[2] = 1
         word2[3] = 0
         embedding_layer = ElmoTokenEmbedder.from_params(vocab=None, params=params)
+        assert embedding_layer.get_output_dim() == 20
+
```
Comment (nelson-liu) [inline allennlp/tests/modules/token_embedders/elmo_token_embedder_test.py]: looks like there's some whitespace on this line that pylint doesn't like.

File: allennlp/tests/modules/token_embedders/elmo_token_embedder_test.py [hunk 2]
```diff
@@ -87,10 +87,12 @@ def test_forward_works_with_projection_layer(self):
         word2[2] = 1
         word2[3] = 0
         embedding_layer = ElmoTokenEmbedder.from_params(vocab=None, params=params)
+        assert embedding_layer.get_output_dim() == 20
+        
         input_tensor = torch.LongTensor([[word1, word2]])
         embedded = embedding_layer(input_tensor).data.numpy()
         assert embedded.shape == (1, 2, 20)
-
+
```
Comment (nelson-liu) [inline allennlp/tests/modules/token_embedders/elmo_token_embedder_test.py]: likewise, looks like there's some whitespace on this line that pylint doesn't like.

File: allennlp/modules/token_embedders/elmo_token_embedder.py [hunk 3]
```diff
@@ -65,11 +65,13 @@ def __init__(self,
...
         if projection_dim:
             self._projection = torch.nn.Linear(self._elmo.get_output_dim(), projection_dim)
+            self.output_dim = projection_dim
         else:
             self._projection = None
-
-    def get_output_dim(self):
-        return self._elmo.get_output_dim()
+            self.output_dim = self._elmo.get_output_dim()
+
```
Comment (nelson-liu) [inline allennlp/modules/token_embedders/elmo_token_embedder.py]: looks like there's some whitespace on this line that pylint doesn't like.

Comment (tingkai-zhang) [reply allennlp/modules/token_embedders/elmo_token_embedder.py]: Thanks! I should have checked that whitespace problem.

Comment (tingkai-zhang) [reply allennlp/modules/token_embedders/elmo_token_embedder.py]: @nelson-liu  Thanks for your review and help!
I think i still failed the pylint test. It gives the following error detail:
```
[18:17:00]Using config file /local/deploy/agent2/work/98197cf33cb401e5/.pylintrc
[18:18:48]Process exited with code 1
```
Is there anything i should fix?

**Comment by nelson-liu** [comment]
i fixed it, see https://github.com/allenai/allennlp/pull/2234/commits/e57707a91d205283cc4a664b1be941e2c3f7b9dc . maybe your editor is inserting these whitespaces for some reason?

**Comment by nelson-liu** [comment]
thanks @tingkai-zhang !

**Comment by tingkai-zhang** [comment]
No worries! I appreciate your help!

**Pull request merged**